src/averageAges.js

```diff
@@ -37,7 +42,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const females = people.filter((person) =>
+    withChildren
+      ? person.sex === 'f'
+        && people.find((child) => person.name === child.mother)
+      : person.sex === 'f');
+
+  const womenAverageAge = females.reduce((acc, person) => (
+    acc + person.died - person.born
+  ), 0) / females.length;
```

Comment:
You use the same callback to calculate the average, let's move it to a separate helper function.